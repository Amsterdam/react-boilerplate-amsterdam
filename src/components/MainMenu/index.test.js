import React from 'react';
import { shallow } from 'enzyme';

import { MainMenu } from './index';

jest.mock('../../signals/incident/containers/IncidentContainer/actions');

describe('<MainMenu />', () => {
  let props;

  beforeEach(() => {
    props = {
      isAuthenticated: false,
      resetIncident: jest.fn()
    };
  });

  describe('rendering', () => {
    it('should render render 1 NavLink components when not authenticated', () => {
      const wrapper = shallow(<MainMenu {...props} />);

      expect(wrapper).toMatchSnapshot();
    });

    it('should render render 2 NavLink components when authenticated', () => {
      props.isAuthenticated = true;
      const wrapper = shallow(<MainMenu {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('should render render 1 NavLink components when not authenticated', () => {
      const wrapper = shallow(<MainMenu {...props} />);
      const event = {
        stopPropagation: jest.fn()
      };

      wrapper.find('NavLink').simulate('click', event);
      expect(event.stopPropagation).toHaveBeenCalled();
      expect(props.resetIncident).toHaveBeenCalled();
    });
  });
});
