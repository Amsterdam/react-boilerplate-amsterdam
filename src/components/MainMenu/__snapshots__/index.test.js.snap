// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainMenu
    isAuthenticated={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container type-nav-primair"
      >
        <nav>
          <ul
            className="links horizontal"
          >
            <li>
              <NavLink
                activeClassName="active"
                to="/"
              >
                <span
                  className="linklabel"
                >
                  <FormattedMessage
                    defaultMessage="Home"
                    id="boilerplate.components.Header.home"
                    values={Object {}}
                  />
                </span>
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                to="/incident"
              >
                <span
                  className="linklabel"
                >
                  <FormattedMessage
                    defaultMessage="Nieuwe melding"
                    id="boilerplate.components.Header.incident"
                    values={Object {}}
                  />
                </span>
              </NavLink>
            </li>
            
          </ul>
        </nav>
      </div>,
      "className": "row main-menu-component",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <ul
            className="links horizontal"
          >
            <li>
              <NavLink
                activeClassName="active"
                to="/"
              >
                <span
                  className="linklabel"
                >
                  <FormattedMessage
                    defaultMessage="Home"
                    id="boilerplate.components.Header.home"
                    values={Object {}}
                  />
                </span>
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                to="/incident"
              >
                <span
                  className="linklabel"
                >
                  <FormattedMessage
                    defaultMessage="Nieuwe melding"
                    id="boilerplate.components.Header.incident"
                    values={Object {}}
                  />
                </span>
              </NavLink>
            </li>
            
          </ul>
        </nav>,
        "className": "container type-nav-primair",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="links horizontal"
          >
            <li>
              <NavLink
                activeClassName="active"
                to="/"
              >
                <span
                  className="linklabel"
                >
                  <FormattedMessage
                    defaultMessage="Home"
                    id="boilerplate.components.Header.home"
                    values={Object {}}
                  />
                </span>
              </NavLink>
            </li>
            <li>
              <NavLink
                activeClassName="active"
                to="/incident"
              >
                <span
                  className="linklabel"
                >
                  <FormattedMessage
                    defaultMessage="Nieuwe melding"
                    id="boilerplate.components.Header.incident"
                    values={Object {}}
                  />
                </span>
              </NavLink>
            </li>
            
          </ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                  activeClassName="active"
                  to="/"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Home"
                      id="boilerplate.components.Header.home"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>,
              <li>
                <NavLink
                  activeClassName="active"
                  to="/incident"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Nieuwe melding"
                      id="boilerplate.components.Header.incident"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>,
              "",
            ],
            "className": "links horizontal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  to="/"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Home"
                      id="boilerplate.components.Header.home"
                      values={Object {}}
                    />
                  </span>
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "children": <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Home"
                      id="boilerplate.components.Header.home"
                      values={Object {}}
                    />
                  </span>,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedMessage
                      defaultMessage="Home"
                      id="boilerplate.components.Header.home"
                      values={Object {}}
                    />,
                    "className": "linklabel",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "defaultMessage": "Home",
                      "id": "boilerplate.components.Header.home",
                      "values": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  to="/incident"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Nieuwe melding"
                      id="boilerplate.components.Header.incident"
                      values={Object {}}
                    />
                  </span>
                </NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "children": <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Nieuwe melding"
                      id="boilerplate.components.Header.incident"
                      values={Object {}}
                    />
                  </span>,
                  "to": "/incident",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedMessage
                      defaultMessage="Nieuwe melding"
                      id="boilerplate.components.Header.incident"
                      values={Object {}}
                    />,
                    "className": "linklabel",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "defaultMessage": "Nieuwe melding",
                      "id": "boilerplate.components.Header.incident",
                      "values": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "li",
            },
            "",
          ],
          "type": "ul",
        },
        "type": "nav",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container type-nav-primair"
        >
          <nav>
            <ul
              className="links horizontal"
            >
              <li>
                <NavLink
                  activeClassName="active"
                  to="/"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Home"
                      id="boilerplate.components.Header.home"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName="active"
                  to="/incident"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Nieuwe melding"
                      id="boilerplate.components.Header.incident"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>
              
            </ul>
          </nav>
        </div>,
        "className": "row main-menu-component",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <nav>
            <ul
              className="links horizontal"
            >
              <li>
                <NavLink
                  activeClassName="active"
                  to="/"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Home"
                      id="boilerplate.components.Header.home"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName="active"
                  to="/incident"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Nieuwe melding"
                      id="boilerplate.components.Header.incident"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>
              
            </ul>
          </nav>,
          "className": "container type-nav-primair",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="links horizontal"
            >
              <li>
                <NavLink
                  activeClassName="active"
                  to="/"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Home"
                      id="boilerplate.components.Header.home"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName="active"
                  to="/incident"
                >
                  <span
                    className="linklabel"
                  >
                    <FormattedMessage
                      defaultMessage="Nieuwe melding"
                      id="boilerplate.components.Header.incident"
                      values={Object {}}
                    />
                  </span>
                </NavLink>
              </li>
              
            </ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <NavLink
                    activeClassName="active"
                    to="/"
                  >
                    <span
                      className="linklabel"
                    >
                      <FormattedMessage
                        defaultMessage="Home"
                        id="boilerplate.components.Header.home"
                        values={Object {}}
                      />
                    </span>
                  </NavLink>
                </li>,
                <li>
                  <NavLink
                    activeClassName="active"
                    to="/incident"
                  >
                    <span
                      className="linklabel"
                    >
                      <FormattedMessage
                        defaultMessage="Nieuwe melding"
                        id="boilerplate.components.Header.incident"
                        values={Object {}}
                      />
                    </span>
                  </NavLink>
                </li>,
                "",
              ],
              "className": "links horizontal",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    to="/"
                  >
                    <span
                      className="linklabel"
                    >
                      <FormattedMessage
                        defaultMessage="Home"
                        id="boilerplate.components.Header.home"
                        values={Object {}}
                      />
                    </span>
                  </NavLink>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "children": <span
                      className="linklabel"
                    >
                      <FormattedMessage
                        defaultMessage="Home"
                        id="boilerplate.components.Header.home"
                        values={Object {}}
                      />
                    </span>,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <FormattedMessage
                        defaultMessage="Home"
                        id="boilerplate.components.Header.home"
                        values={Object {}}
                      />,
                      "className": "linklabel",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "defaultMessage": "Home",
                        "id": "boilerplate.components.Header.home",
                        "values": Object {},
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "span",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavLink
                    activeClassName="active"
                    to="/incident"
                  >
                    <span
                      className="linklabel"
                    >
                      <FormattedMessage
                        defaultMessage="Nieuwe melding"
                        id="boilerplate.components.Header.incident"
                        values={Object {}}
                      />
                    </span>
                  </NavLink>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "active",
                    "children": <span
                      className="linklabel"
                    >
                      <FormattedMessage
                        defaultMessage="Nieuwe melding"
                        id="boilerplate.components.Header.incident"
                        values={Object {}}
                      />
                    </span>,
                    "to": "/incident",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <FormattedMessage
                        defaultMessage="Nieuwe melding"
                        id="boilerplate.components.Header.incident"
                        values={Object {}}
                      />,
                      "className": "linklabel",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "defaultMessage": "Nieuwe melding",
                        "id": "boilerplate.components.Header.incident",
                        "values": Object {},
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "span",
                  },
                  "type": [Function],
                },
                "type": "li",
              },
              "",
            ],
            "type": "ul",
          },
          "type": "nav",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
